modules/_labs/tabs/react/lib/Tab.tsx

```diff
@@ -71,59 +88,36 @@ const StyledButton = styled('button')<{isSelected: boolean}>(
   }
 );
 
-const Tab = ({name = '', children, ...elemProps}: TabProps) => {
-  const {
-    id,
-    intentTab,
-    resetIntentTab,
-    activeTab,
-    setActiveTab,
-    setSelectedTabRect,
-    registerTab,
-    unregisterTab,
-  } = useTab();
-  const tabRef = React.useRef<HTMLButtonElement>(null);
-  const [tabName, setTabName] = React.useState(name);
+export const Tab = createComponent('button')({
+  displayName: 'Tabs.Item',
+  Component: ({name = '', model, children, ...elemProps}: TabProps, ref, Element) => {
+    const {localRef, elementRef} = useLocalRef(ref);
+    const {state, events} = useModelContext(TabsModelContext, model);
 
-  // useLayoutEffect because we don't want to render with incorrect ID
-  React.useLayoutEffect(() => {
-    const tabElement = tabRef.current as HTMLElement;
-    const tabName = registerTab(tabElement, name);
-    setTabName(tabName);
+    const tabName = useRegisterItem({state, events}, localRef, name);
 
-    return () => {
-      unregisterTab(tabElement);
+    const onSelect = () => {
+      events.activateTab({tab: tabName});
     };
-  }, [name, registerTab, unregisterTab]);
 
-  const onSelect = () => {
-    setActiveTab(tabName);
-  };
+    const isSelected = !!tabName && state.activeTab === tabName;
 
-  const isSelected = !!tabName && activeTab === tabName;
+    const props = mergeProps(
+      {
+        id: `tab-${state.id}-${tabName}`,
+        tabIndex: !!tabName && state.currentId === tabName ? undefined : -1,
+        'aria-selected': isSelected ? true : undefined,
+        'aria-controls': `tabpanel-${state.id}-${tabName}`,
+        onClick: onSelect,
+        isSelected: isSelected,
+      },
+      elemProps
+    );
```

Comment:
Every prop list that uses a callback needs to use `mergeProps` to make sure callbacks are handled properly. This example could have use only the `onClick` instead of all the other props... Not 100% on this one.